client/FxAccountClient.js

```diff
@@ -1818,6 +1818,64 @@ define([
       });
   };
 
+  /**
+   * Replace user's recovery codes.
+   *
+   * @method replaceRecoveryCodes
+   * @param {String} sessionToken SessionToken obtained from signIn
+   */
+  FxAccountClient.prototype.replaceRecoveryCodes = function (sessionToken) {
+    var request = this.request;
+    return Promise.resolve()
+      .then(function () {
+        required(sessionToken, 'sessionToken');
+
+        return hawkCredentials(sessionToken, 'sessionToken', HKDF_SIZE);
+      })
+      .then(function (creds) {
+
+        return request.send('/recoveryCodes', 'GET', creds);
+      });
+  };
+
+  /**
+   * Consume recovery code.
+   *
+   * @method consumeRecoveryCode
+   * @param {String} sessionToken SessionToken obtained from signIn
+   * @param {String} code recovery code
+   * @param {Object} [options.metricsContext={}] Metrics context metadata
+   *   @param {String} options.metricsContext.deviceId identifier for the current device
+   *   @param {String} options.metricsContext.flowId identifier for the current event flow
+   *   @param {Number} options.metricsContext.flowBeginTime flow.begin event time
+   *   @param {Number} options.metricsContext.utmCampaign marketing campaign identifier
+   *   @param {Number} options.metricsContext.utmContent content identifier
+   *   @param {Number} options.metricsContext.utmMedium acquisition medium
+   *   @param {Number} options.metricsContext.utmSource traffic source
+   *   @param {Number} options.metricsContext.utmTerm search terms
+   */
+  FxAccountClient.prototype.consumeRecoveryCode = function (sessionToken, code, options) {
+    var request = this.request;
+    return Promise.resolve()
+      .then(function () {
+        required(sessionToken, 'sessionToken');
+        required(code, 'code');
+
+        return hawkCredentials(sessionToken, 'sessionToken', HKDF_SIZE);
+      })
+      .then(function (creds) {
+        var data = {
+          code: code
+        };
+
+        if (options && options.metricsContext) {
+          data.metricsContext = metricsContext.marshall(options.metricsContext);
```

Comment:
I don't see `metricsContext` in the payload for this endpoint on the auth server. Won't it cause the request to fail if present? Should we remove here (and in the comment above), or is there a PR coming to add it there for some reason?

Reply:
Nice catch! Opened https://github.com/mozilla/fxa-auth-server/pull/2367 to fix this. 

> Won't it cause the request to fail if present?

Turns out that it would only fail if a key was invalid and it would also strip out unknown keys. Requests didn't fail on content-server because I wasn't passing it in there. Will update that in https://github.com/mozilla/fxa-content-server/pull/5992.